### PR TITLE
Do not fail in the debug middleware

### LINKED ISSUE #1976: Sample app with Prometheus metrics and debug middleware broke on 0.0.3 -> 0.0.4 update
**Describe the bug**
I have a sample application that consists of some demo routes and a route exposing metrics to Prometheus (similar to https://zio.dev/guides/tutorials/monitor-a-zio-application-using-zio-built-in-metric-system/).

I also use the cors, timeout and debug middlewares. After updating to zio-http 0.0.4, I started getting the error below but If I **remove the debug middleware**, the application runs fine.

Error:

```
❯ ./mill backend.run
[35/48] backend.compile
[info] compiling 1 Scala source to /Users/cdepaula/repos/scala-playground/zio-scalajs-stack/out/backend/compile.dest/classes ...
[error] -- [E007] Type Mismatch Error: /Users/cdepaula/repos/scala-playground/zio-scalajs-stack/backend/src/Main.scala:40:11
[error] 40 |    .serve(httpRoutes)
[error]    |           ^^^^^^^^^^
[error]    |  Found:    (com.carlosedp.zioscalajs.backend.Main.httpRoutes :
[error]    |    zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,
[error]    |      java.io.IOException, zio.http.Request, zio.http.Response]
[error]    |  )
[error]    |  Required: zio.http.App[Any]
[error]    |----------------------------------------------------------------------------
[error]    | Explanation (enabled by `-explain`)
[error]    |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[error]    |
[error]    | Tree: com.carlosedp.zioscalajs.backend.Main.httpRoutes
[error]    | I tried to show that
[error]    |   (com.carlosedp.zioscalajs.backend.Main.httpRoutes :
[error]    |   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,
[error]    |     java.io.IOException, zio.http.Request, zio.http.Response]
[error]    | )
[error]    | conforms to
[error]    |   zio.http.App[Any]
[error]    | but the comparison trace ended with `false`:
[error]    |
[error]    |   ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.App[Any]
[error]    |     ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.HttpApp[Any, zio.http.Response]
[error]    |       ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response]
[error]    |         ==> zio.http.type  <:  zio.http.type
[error]    |         <== zio.http.type  <:  zio.http.type = true
[error]    |         ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]    |           ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]    |           <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]    |         <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]    |         ==> zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,   java.io.IOException, zio.http.Request, zio.http.Response]  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] (left is approximated)
[error]    |           ==> zio.http.type  <:  zio.http.type
[error]    |           <== zio.http.type  <:  zio.http.type = true
[error]    |           ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]    |             ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]    |             <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]    |           <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]    |         <== zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,   java.io.IOException, zio.http.Request, zio.http.Response]  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] (left is approximated) = false
[error]    |       <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] = false
[error]    |     <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.HttpApp[Any, zio.http.Response] = false
[error]    |   <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.App[Any] = false
[error]    |
[error]    | The tests were made under a constraint with:
[error]    |  uninstantiated variables: R, E1
[error]    |  constrained types:
[error]    |   [R]
[error]    |     (httpApp: zio.http.App[R], errorCallback:
[error]    |       Option[zio.http.Server.ErrorCallback])
[error]    |       (implicit trace: zio.Trace): zio.URIO[R & zio.http.Server, Nothing]
[error]    |   ,
[error]    |   [R]
[error]    |     (httpApp: zio.http.App[R], errorCallback:
[error]    |       Option[zio.http.Server.ErrorCallback])
[error]    |       (implicit trace: zio.Trace): zio.URIO[R & zio.http.Server, Nothing]
[error]    |   ,
[error]    | [E1](layer: zio.ZLayer[?, E1, ?]*): zio.ZIO[Any, E1, Nothing]
[error]    |  bounds:
[error]    |      R
[error]    |      R := Any
[error]    |      E1
[error]    |  ordering:
[error]    |  co-deps:
[error]    |  contra-deps:
[error]     ----------------------------------------------------------------------------
[error] Explanation
[error] ===========
[error]
[error] Tree: com.carlosedp.zioscalajs.backend.Main.httpRoutes
[error] I tried to show that
[error]   (com.carlosedp.zioscalajs.backend.Main.httpRoutes :
[error]   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,
[error]     java.io.IOException, zio.http.Request, zio.http.Response]
[error] )
[error] conforms to
[error]   zio.http.App[Any]
[error] but the comparison trace ended with `false`:
[error]
[error]   ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.App[Any]
[error]     ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.HttpApp[Any, zio.http.Response]
[error]       ==> (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response]
[error]         ==> zio.http.type  <:  zio.http.type
[error]         <== zio.http.type  <:  zio.http.type = true
[error]         ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]           ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]           <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]         <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]         ==> zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,   java.io.IOException, zio.http.Request, zio.http.Response]  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] (left is approximated)
[error]           ==> zio.http.type  <:  zio.http.type
[error]           <== zio.http.type  <:  zio.http.type = true
[error]           ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]             ==> Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher
[error]             <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]           <== Any  <:  zio.metrics.connectors.prometheus.PrometheusPublisher = false
[error]         <== zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,   java.io.IOException, zio.http.Request, zio.http.Response]  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] (left is approximated) = false
[error]       <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.Http[Any, zio.http.Response, zio.http.Request, zio.http.Response] = false
[error]     <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.HttpApp[Any, zio.http.Response] = false
[error]   <== (com.carlosedp.zioscalajs.backend.Main.httpRoutes :   zio.http.Http[zio.metrics.connectors.prometheus.PrometheusPublisher,     java.io.IOException, zio.http.Request, zio.http.Response] )  <:  zio.http.App[Any] = false
[error]
[error] The tests were made under a constraint with:
[error]  uninstantiated variables: R, E1
[error]  constrained types:
[error]   [R]
[error]     (httpApp: zio.http.App[R], errorCallback:
[error]       Option[zio.http.Server.ErrorCallback])
[error]       (implicit trace: zio.Trace): zio.URIO[R & zio.http.Server, Nothing]
[error]   ,
[error]   [R]
[error]     (httpApp: zio.http.App[R], errorCallback:
[error]       Option[zio.http.Server.ErrorCallback])
[error]       (implicit trace: zio.Trace): zio.URIO[R & zio.http.Server, Nothing]
[error]   ,
[error] [E1](layer: zio.ZLayer[?, E1, ?]*): zio.ZIO[Any, E1, Nothing]
[error]  bounds:
[error]      R
[error]      R := Any
[error]      E1
[error]  ordering:
[error]  co-deps:
[error]  contra-deps:
[error] one error found
1 targets failed
backend.compile Compilation failed
```

Sample code can be seen at: https://github.com/carlosedp/zio-scalajs-stack/blob/17f9596eee72285179fab1cc40c78134311817d1/backend/src/Main.scala#L25 and the metrics route at https://github.com/carlosedp/zio-scalajs-stack/blob/main/backend/src/MetricsApp.scala but as an overview I have:

```scala

object MetricsApp {
  def apply(): Http[PrometheusPublisher, Nothing, Request, Response] =
    Http.collectZIO[Request] { case Method.GET -> !! / "metrics" =>
      ZIO.serviceWithZIO[PrometheusPublisher](
        _.get.map(Response.text),
      ) @@ httpHitsMetric("GET", "/metrics")
    }
}

# and in my main:

...
  // Add routes and middleware
  val httpRoutes = (MetricsApp() ++ HomeApp() ++ GreetingApp()) @@
    Middleware.cors(corsConfig) @@
    Middleware.metrics(MetricsApp.pathLabelMapper) @@
    Middleware.timeout(5.seconds) 
	@@ Middleware.debug // <- Commenting this works

  // Define ZIO-http server
  val server: ZIO[Any, Throwable, Nothing] = Server
    .serve(httpRoutes)
    .provide( // Add required layers
      ServerConfig.live(config),
      Server.live,
      publisherLayer,
      prometheusLayer,
      ZLayer.succeed(MetricsConfig(200.millis)), // Metrics pull interval from internal store
    )

```

**Expected behaviour**
The server to work as in 0.0.3.

### PR DESCRIPTION
**Pull request by vigoo**
Resolves #1976

### CODE REVIEW

File: zio-http/src/main/scala/zio/http/middleware/RequestHandlerMiddlewares.scala [hunk 1]
```diff
@@ -37,16 +37,17 @@ private[zio] trait RequestHandlerMiddlewares
...
+    new RequestHandlerMiddleware[Any, Nothing] {
+      override def apply[R1 <: Any, Err1 >: Nothing](
         handler: Handler[R1, Err1, Request, Response],
       )(implicit trace: Trace): Handler[R1, Err1, Request, Response] =
         Handler.fromFunctionZIO { request =>
           handler.runZIO(request).timed.flatMap { case (duration, response) =>
             Console
               .printLine(s"${response.status.code} ${request.method} ${request.url.encode} ${duration.toMillis}ms")
               .as(response)
+              .orDie
```
Comment (adamgfraser) [inline zio-http/src/main/scala/zio/http/middleware/RequestHandlerMiddlewares.scala:50]: Could also just do `ZIO.debug` here.

**Review by adamgfraser — APPROVED** [review]

**Comment by vigoo** [comment]
Changed to `ZIO.debug`

**Review by adamgfraser — APPROVED** [review]

**Comment by vigoo** [comment]
Seems like it was tested with TestConsole, I’ll revert

**Comment by adamgfraser** [comment]
👍

**Pull request merged**